pkg/registry/authorization/selfsubjectaccessreview/rest.go

```diff
@@ -72,6 +76,33 @@ func (r *REST) Create(ctx context.Context, obj runtime.Object, createValidation
 		}
 	}
 
+	hasUserFull, hasUserCheckAccess := false, false
+	for _, scope := range userFromCtx.GetExtra()[authorizationv1.ScopesKey] {
+		switch scope {
+		case authorizationscope.UserFull:
+			hasUserFull = true
+			break
+
+		case authorizationscope.UserAccessCheck:
+			hasUserCheckAccess = true
+		}
+	}
+
+	var userToCheck user.Info
+	if hasUserCheckAccess && !hasUserFull {
+		userToCheck = &user.DefaultInfo{
+			Name:   userFromCtx.GetName(),
+			UID:    userFromCtx.GetUID(),
+			Groups: userFromCtx.GetGroups(),
+			Extra: map[string][]string{
+				authorizationv1.ScopesKey: {authorizationscope.UserFull},
+			},
+		}
+
+	} else {
+		userToCheck = userFromCtx
+	}
```

Comment:
```go
    userExtraCopy := make(map[string][]string{})
    for k, v := range userFromCtx.GetExtra() {
        userExtraCopy[k] = v
    }
        
    for _, scope := range userExtraCopy[authorizationv1.ScopesKey] {
        switch scope {
            case authorizationscope.UserFull:
                break

            case authorizationscope.UserAccessCheck:
                userExtraCopy[authorizationv1.ScopesKey] = []string{authorizationscope.UserFull}
                break
        }
    }
	
	userToCheck :=  &user.DefaultInfo{
        Name:   userFromCtx.GetName(),
        UID:    userFromCtx.GetUID(),
        Groups: userFromCtx.GetGroups(),
        Extra: userExtraCopy,
    }
```

Reply:
Simplified code as per suggestion; I also removed the check for UserFull to simplify the code further.